roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -114,14 +128,18 @@ impl JDsMempool {
         Ok(())
     }
 
-    pub fn to_short_ids(&self, nonce: u64) -> Option<HashMap<[u8; 6], Transaction>> {
+    pub fn to_short_ids(&self, nonce: u64) -> Option<HashMap<[u8; 6], TransactionWithHash>> {
```

Comment:
The struct `TransactionWithHash` was introduced on to represent transactions in the mempool, but you changed this representation. Did you consider to get rid of it?
You can return `Option<HashMap<[u8; 6], (Txid, Transaction)>>`.